### ISSUE TITLE
GFPGAN to TorchScript/TensorRT

### ISSUE THREAD

**Issue by lschaupp** [TencentARC/GFPGAN#275]
Hello, I am trying to convert the GFPGAN model to TorchScript/TensorRT to increase model performance. Has there be made any efforts yet on this?

**Comment by NingNanXin** [comment]
Hi, I successfully export trt with TensorRt 8.5, but onnx result !=  tensorrt result.
Did you meet this problem?
This my polygraph output.
<img width="939" alt="image" src="https://user-images.githubusercontent.com/56398622/229459760-40babdd3-6d68-40c0-93c6-356374c36a80.png">

**Comment by bychen7** [comment]
@NingNanXin Hi, you can try this https://github.com/bychen7/Face-Restoration-TensorRT.

**Comment by topninja** [comment]
@NingNanXin 
https://github.com/bychen7/Face-Restoration-TensorRT
This repo can be run on Windows 10?

**Comment by bychen7** [comment]
> @NingNanXin https://github.com/bychen7/Face-Restoration-TensorRT This repo can be run on Windows 10?

I think it is possible, and the code does not need to be changed. However, I have not tested it on Win10, and it would require replacing all dependencies with their Windows versions.

**Comment by NingNanXin** [comment]
> @NingNanXin Hi, you can try this https://github.com/bychen7/Face-Restoration-TensorRT.

Thanks for your work, I think you can push your work in Tensorrtx by wangxinyu, thus more people will benefit from your work.
I test the work in ubuntu18.04, trt8.2, and got a good performance, but you didn't set fp16 precision, so all of your onnx keep fp32?

<img width="297" alt="image" src="https://github.com/TencentARC/GFPGAN/assets/56398622/056ab074-b5b0-408c-b58b-2261fb328bde">

And you can try trt>8.5, it can deal with the two inputs of a convolution

**Comment by bychen7** [comment]
> > @NingNanXin Hi, you can try this https://github.com/bychen7/Face-Restoration-TensorRT.
> 
> Thanks for your work, I think you can push your work in Tensorrtx by wangxinyu, thus more people will benefit from your work. I test the work in ubuntu18.04, trt8.2, and got a good performance, but you didn't set fp16 precision, so all of your onnx keep fp32?
> 
> <img alt="image" width="297" src="https://user-images.githubusercontent.com/56398622/239112773-056ab074-b5b0-408c-b58b-2261fb328bde.png">
> And you can try trt>8.5, it can deal with the two inputs of a convolution

Thank you for your suggestion. The quality of fp16 generation may decrease. This work was done a year ago, and I reviewed the records from that time. Please refer to the following comparison results.
<img width="782" alt="exp" src="https://github.com/TencentARC/GFPGAN/assets/55865490/304505ac-48da-481f-8478-8c4b38c8e61e">

**Comment by NingNanXin** [comment]
@bychen7 
Yes, I convert GFPGAN to fp16 and got the same problem. Even if I retrain the model the problem cannot be solved.  However, GPEN can right convert to fp16.
Another problem do you find the trt infer speed is slower than torch when the resolution is 512*512. For example, GFPGAN jit model got an average 14ms/frame, but fp32 trt got an average 36ms/frame, my test device is RTX2080ti.

**Comment by debasishaimonk** [comment]
@NingNanXin did u solve it?

**Comment by NingNanXin** [comment]
@debasishaimonk No, I try to keep some layers as fp32,others fp16, but trt which export by this way did not work
